src/fides/api/util/tcf/tc_string.py

```diff
@@ -180,3 +188,87 @@ def build_disclosed_vendors_string(model: TCModel) -> str:
     return base64.urlsafe_b64encode(
         _convert_bitstring_to_bytes(disclosed_vendor_bits)
     ).decode()
+
+
+def boolean_to_user_consent_preference(preference: bool) -> UserConsentPreference:
+    return UserConsentPreference.opt_in if preference else UserConsentPreference.opt_out
+
+
+FidesPreferenceType = Union[TCFPurposeSave, TCFVendorSave, TCFSpecialFeatureSave]
+
+
+def convert_to_fides_preference(
+    datamap_options: List[int],
+    tc_string_preferences: Dict[int, bool],
+    preference_class: Union[
+        Type[TCFPurposeSave], Type[TCFVendorSave], Type[TCFSpecialFeatureSave]
+    ],
+) -> List[FidesPreferenceType]:
+    """For a field in the TC string, convert to a list of preferences where we opt-in if it is included in the string,
+    and opt-out if it is not in the string, but in the datamap"""
+    # Loop through all the datamap options that are currently available.  For example, all of the vendors
+    # from the datamap against which we can save preferences with a consent legal basis
+    preferences_array: List[FidesPreferenceType] = []
+
+    for identifier in datamap_options:
+        # Check if there's an opt_in encoded in the string.  Otherwise, we assume the user is opting out.
+        preference: bool = tc_string_preferences.get(identifier, False)
+
+        if isinstance(preference_class, TCFVendorSave):
+            # Vendors are currently saved as strings in our db
+            identifier = str(identifier)
+
+        fides_preference: FidesPreferenceType = preference_class(
+            id=identifier, preference=boolean_to_user_consent_preference(preference)
+        )
+        preferences_array.append(fides_preference)
+    return preferences_array
+
+
+def decode_tc_string_to_preferences(
```

Comment:
This method decodes the TC string and formats it the exact same way I'd expect to receive it if someone was passing in privacy preferences via the API.  I then replace the incoming request body with these unpacked preferences, and can save privacy preferences as normal.